### PR TITLE
Save data to DynamoDB

### PR DESCRIPTION
**Pull request by FenrirUnbound**
Save a given data item into a specified DynamoDB table

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -63,8 +63,24 @@ class Dynamodb extends Datastore {
...
     }
 
+    /**
+     * Save a item in the specified DynamoDB table
+     * @param  {Object}   config             Configuration object
+     * @param  {String}   config.table       Table name
+     * @param  {Object}   config.params      Record data
+     * @param  {String}   config.params.id   Unique id. Typically the desired primary key
+     * @param  {Object}   config.params.data The data to save
+     * @param  {Function} callback           fn(err)
```
Comment (d2lam) [inline index.js]: fn(err,data) to be consistent with imdb: 
https://github.com/screwdriver-cd/screwdriver-datastore-imdb/blob/master/index.js#L67

File: index.js [hunk 2]
```diff
@@ -63,8 +63,24 @@ class Dynamodb extends Datastore {
...
+     *                                       err - Error object
+     */
     save(config, callback) {
-        callback(null);
+        const id = config.params.id;
+        const userData = config.params.data;
+        const client = this.client[config.table];
```
Comment (nkatzman) [inline index.js:80]: Do you need to check whether the client exists?

Comment (FenrirUnbound) [inline index.js:80]: Checked ✔️

File: index.js (hunk 1 above)
Comment (FenrirUnbound) [inline index.js]: Addressed ✉️

File: README.md [hunk 3]
```diff
@@ -9,6 +9,97 @@
...
+Save a record in the datastore. Returns saved data.
+
+**Arguments**
+
+* `config` - An `object`. Each of its properties defines your save operation
+* `config.table` - A `string`. The datastore table name
+* `config.params` - An object. Each of its properties defines the save parameters
+* `config.params.id` - A `string`. The ID to associate the data with
+* `config.params.data` - An object. This is what will be saved in the datastore
+* `callback()` - A callback which is called when the task is completed
```
Comment (d2lam) [inline README.md]: callback(err, data) ....

File: index.js (hunk 1 above)
Comment (d2lam) [inline index.js]: still need to change `fn(err)` to `fn(err,data)` in the comment

File: index.js [hunk 4]
```diff
@@ -63,8 +63,37 @@ class Dynamodb extends Datastore {
...
+     *                                       data - Data saved in the table
+     */
     save(config, callback) {
-        callback(null);
+        const id = config.params.id;
+        const userData = config.params.data;
+        const client = this.client[config.table];
+
+        if (!client) {
+            const err = new Error(`invalid table name "${config.table}"`);
```
Comment (d2lam) [inline index.js]: Sorry for being picky but can we upper case the letter `i` :p

**Comment by d2lam** [comment]
Small comments. The rest LGTM 👍

**Comment by nkatzman** [comment]
👍

**Pull request merged**